iocore/hostdb/HostDB.cc

```diff
@@ -1593,29 +1437,21 @@ HostDBContinuation::dnsEvent(int event, HostEnt *e)
               }
             }
           }
-        } else {
-          for (ii = 0; ii < nn; ++ii) {
-            if (is_addr_valid(af, e->ent.h_addr_list[ii])) {
-              HostDBInfo &item = rr_data->info[i];
-              memset(&item, 0, sizeof(item));
-              ip_addr_set(item.ip(), af, e->ent.h_addr_list[ii]);
-              item.full = 1;
-              item.round_robin = 0;
-              item.round_robin_elt = 1;
-              item.reverse_dns = 0;
-              item.is_srv = 0;
-              item.md5_high = r->md5_high;
-              item.md5_low = r->md5_low;
-              item.md5_low_low = r->md5_low_low;
-              item.hostname_offset = 0;
-              if (!restore_info(&item, old_r, old_info, old_rr_data)) {
-                item.app.allotment.application1 = 0;
-                item.app.allotment.application2 = 0;
-              }
-              ++i;
+        } else { // Otherwise this is a regular dns response
+          for (int i = 0; i < valid_records; ++i) {
+            HostDBInfo &item = rr_data->info[i];
+            memset(&item, 0, sizeof(item));
+            ip_addr_set(item.ip(), af, e->ent.h_addr_list[i]);
+            item.round_robin = 0;
```

Comment:
Well, first off-- although this is in the diff, that is actually what it is today.

From looking at it the field is actually `unsigned int round_robin : 1;` so technically it is an int ;)